impl/src/main/java/org/jboss/weld/bootstrap/events/AbstractDefinitionContainerEvent.java

```diff
@@ -43,7 +44,12 @@ public void addDefinitionError(Throwable t) {
     public void fire() {
         super.fire();
         if (!getErrors().isEmpty()) {
-            throw new DefinitionException(getErrors());
+            if (getErrors().size() == 1 && getErrors().get(0) instanceof DeploymentException) {
+                // if the throwable was deployment exception, rethrow that
+                throw new DeploymentException(getErrors().get(0));
```

Comment:
Hm, shouldn't this be more like `throw getErrors().get(0)`? I mean you don't really rethrow but wrap the original exception...

Reply:
Done